src/core/providers/ogr/qgsogrproviderutils.cpp

```diff
@@ -2565,7 +2570,8 @@ QList< QgsProviderSublayerDetails > QgsOgrProviderUtils::querySubLayerList( int
       // in the uri for the sublayers (otherwise we'll be forced to re-do this iteration whenever
       // the uri from the sublayer is used to construct an actual vector layer)
       if ( details.wkbType() != QgsWkbTypes::Unknown )
-        parts.insert( QStringLiteral( "geometryType" ), ogrWkbGeometryTypeName( countIt.key() ) );
+        parts.insert( QStringLiteral( "geometryType" ),
```

Comment:
this looks good to me but looking a bit at tests, it is not obvious that a URI with a geometrytype set to a non-2D value is tested & works (I may have missed it). It could be appropriate perhaps in the test you modified to open the layer through the uri and check that features are returned

Reply:
Looks good to me too -- the query sublayers functionality is heavily soaked in tests, so there's little chance of a regression there.